### ISSUE TITLE
function return type

### ISSUE THREAD

**Issue by benzino77** [marcoschwartz/aREST#73]
Hi Marco,
Thanks for the great work!

In this piece of code below there is uint8_t as a return value of the user defined function. Shouldn't be rather:
**int result = functions\[value\](arguments);** 
as user defined functions needs to take a String as the unique argument and return an **integer**

    // Function selected
    if (command == 'f') {

    // Execute function
    ---> uint8_t result = functions[value](arguments);

    // Send feedback to client
    if (!LIGHTWEIGHT) {
     addToBuffer(F("{\"return_value\": "));
     addToBuffer(result);
     addToBuffer(F(", "));
     //addToBuffer(F(", \"message\": \""));
     //addToBuffer(functions_names[value]);
     //addToBuffer(F(" executed\", "));
    }
    }

**Comment by marcoschwartz** [comment]
Hello,

Thanks for using aREST! Basically the return code is just here to return 0 or 1 (failure or success), so it's a bit an overkill to use an int variable here, to save memory space. Does that answer your question?

**Comment by benzino77** [comment]
That's what I thought. I just couldn't figure out why my function returns values from 0-255 when it should return "full" int, until I found that it is mapped to uint8_t.
I've changed aREST.h on my side, because I need to have "full" int returned by user defined functions (it doesn't change much from the memory perspective because I have only 3 functions defined in my scetch).
Thanks for your answer!
